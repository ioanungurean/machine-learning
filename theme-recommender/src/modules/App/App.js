import React, { Component } from 'react';

import Header from 'components/Header/Header';
import ThemeRating from 'components/ThemeRating/ThemeRating';
import ThemeList from 'components/ThemeList/ThemeList';

import { predictThemes } from "utils";

import './App.scss';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      themes: [],
    };

    this.getNewPredictions = this.getNewPredictions.bind(this);
  }

  componentDidMount() {
    this.setState({
      themes: predictThemes(),
    });
  }

  getNewPredictions() {
    this.setState({
      themes: predictThemes(),
    });
  }

  render() {
    return (
      <div className="app">
        <Header />

        <div className="app_main">
          <ThemeRating getNewPredictions={this.getNewPredictions}/>
          <ThemeList themes={this.state.themes}/>
        </div>
      </div>
    );
  }
}

export default App;
