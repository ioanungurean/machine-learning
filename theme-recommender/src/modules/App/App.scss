@import 'shared.scss';
@import 'sizes.scss';

.app {
  display: flex;
  flex-direction: column;

  .app_main {
    display: flex;
    flex-direction: column;
    align-items: center;

    padding: map-get($sizes, big);

    > * {
      margin-bottom: map-get($sizes, big);
    }
  }
}
