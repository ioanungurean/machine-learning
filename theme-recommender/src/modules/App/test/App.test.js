import React from 'react';
import { shallow } from 'enzyme';

import App from '../App';

describe('<App />', () => {
  test('should render without throwing an error', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});
