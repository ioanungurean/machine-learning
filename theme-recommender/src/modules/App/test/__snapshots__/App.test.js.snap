// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render without throwing an error 1`] = `
<div
  className="app"
>
  <div
    className="header"
  >
    <div
      className="icon-react"
    />
    <a
      href="#"
    >
      Home
    </a>
    <a
      href="https://github.com/ioanungurean/forge-react-app"
    >
      <Title
        title="Forge React App on Github"
      />
    </a>
  </div>
</div>
`;
